article-analyzer/src/main/java/co/inventorsoft/academy/spring/service/CategoryService.java

```diff
@@ -0,0 +1,30 @@
+package co.inventorsoft.academy.spring.service;
+
+import org.springframework.stereotype.Service;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+public class CategoryService {
+    public Set<String> getCategories(List<List<String>> splitWords) {
```

Comment:
love it, simple and does the job, but almost correctly

returns `["websocket"]` to me

the fix would be to move everything into the for loop on line 17 (even the map before) and add all found words in each article separately into that result set

Reply:
fixed